ansible/ansible_collections/agentic/agent_configs/plugins/module_utils/agent_config_common.py

```
@@ -434,24 +411,145 @@ def manage_named_toml_entry(module, path: str, root_key: str, name: str, desired
+
+def _get_hook_list(
+    module,
+    group: Dict[str, Any],
+    event: str,
+    path: str,
+    *,
+    create: bool,
+) -> List[Any]:
+    hook_list = group.setdefault("hooks", []) if create else group.get("hooks", [])
+    if not isinstance(hook_list, list):
+        module.fail_json(msg="Expected hooks list under event %s in %s" % (event, path))
```

Ruff (0.15.11)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)